### PR TITLE
fix(assistant): make surface completion writes deterministic and consistent on failure

### PR DESCRIPTION
**Pull request by Jasonnnz**
## Summary
- Drop the prelocated-hit optimization so the completion write always targets the same row regardless of debounce timing
- Roll back the in-memory completion patch when the persisted write fails, so memory and the database cannot disagree
- Keep live surface data when falling back to persisted state, and keep the guardian withdrawal broadcast on a transient write failure
- Persist before announcing on the stale-surface dismiss sweep
- Add a real-database test covering a surface behind the context-compaction boundary

Part of plan: choice-surface-completion-persist.md (self-review remediation, round 2)

### CODE REVIEW

File: assistant/src/__tests__/surface-completion-compaction-boundary.test.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+const GUARDIAN = { requesterCanAccessMemory: true };
+
+/**
+ * Four real message rows, the oldest carrying a `choice` surface between two
+ * text siblings, plus a conversation compacted past that row.
+ */
+function seedCompactedConversation(): void {
+  const db = getDb();
+  db.delete(messages).run();
+  db.delete(conversations).run();
```
[the remark this answers is not in the document]
Comment (Jasonnnz) [reply assistant/src/__tests__/surface-completion-compaction-boundary.test.ts:34]: Addressed in direct push to run-plan/surface-completion/fix-r2-1

**Comment by Jasonnnz** [comment]
@codex review this PR again — the previous issues have been fixed in commit f2d46f3eb4fa1d449825aefc2800807ddec9a259

**Comment by Jasonnnz** [comment]
@devin review this PR again — the previous issues have been fixed in commit f2d46f3eb4fa1d449825aefc2800807ddec9a259
